docs-src/0.6/src/reference/component_props.md

```diff
@@ -151,3 +151,185 @@ DemoFrame {
+
+```rust, no_run
+{{#include src/doc_examples/component_reactive_props.rs:ReactiveTemperature}}
+```
+
+The `use_memo` hook now **subscribes** to the signal. Whenever `celsius` changes, the memo is recalculated.
+
+### `use_reactive!` for Precision Control
+
+If you prefer to avoid signals, use [`use_reactive`](https://docs.rs/dioxus/latest/dioxus/prelude/macro.use_reactive.html) macro to **declare dependencies** manually inside hooks:
+
+```rust, no_run
+{{#include src/doc_examples/component_reactive_props.rs:UseReactive}}
+```
+
+This gives you **deterministic control** over which variables should cause recomputation, useful for deeply nested computations or external state systems.
+
+### Gotchas and Tips
+
+- **Signals are both `Copy`able and `Clone`**able: You can pass `ReadOnlySignal<T>` by value, reference, or clone it. Internally, it's a lightweight handle to a reactive state. This makes it easy to lift signals into other hooks without worrying about ownership issues.
+
+- **You can combine signals with the `use_reactive!` macro**: This is useful when only part of your dependencies are reactive. For example:
+
+  ```rust, no_run
+  {{#include src/doc_examples/component_reactive_props.rs:UseReactiveGotcha}}
+  ```
+
+- **Avoid using non-reactive props in `use_future` or `use_resource`**: These hooks only re-run when their dependencies change. If those dependencies are non-reactive (like `String` or `f64`), the hook won't re-execute when they update. Example of incorrect usage:
+
+  ```rust, no_run
+  {{#include src/doc_examples/component_reactive_props.rs:UseFutureGotcha}}
+  ```
+
+  Correct approach using a reactive prop:
+
+  ```rust, no_run
+  {{#include src/doc_examples/component_reactive_props.rs:UseFutureCorrect}}
+  ```
+
+- **Memoized values from `use_memo` are reactive**: The result of `use_memo` is a value. If you need the result to drive other reactive state, store it in a memo:
+
+  ```rust, no_run
+  // ✅ This is a reactive memo that will update when `count` changes
+  let doubled = use_memo(move || count() * 2);
+  ```
+
+- **`use_signal` vs. `use_memo`**: Use `use_signal` when the value will change due to internal component state or user interaction. Use `use_memo` when deriving a value from other reactive props or signals and you want automatic recomputation.
+
+- **Signal updates are conditional on value changes**: If you call `.set()` on a signal with the same value it already holds, dependent effects and hooks will not re-run. If you need to force an update, consider mutating inner data if you're using something like `Signal<Vec<T>>`.
+
+- **Cloning signals in every render can be inefficient**: Although cloning signals is cheap, avoid doing it repeatedly inside `rsx!` or render loops. Instead, just read the signal:
+
+  ```rust, no_run
+  rsx! {
+      div { "Value: {count()}" }
```

Comment:
This clones the value inside the signal instead of the pointer, which is much more expensive

Reply:
It depends on the value inside the signal, but generally, the second code snippet is cheaper.

This will clone the value inside the signal, not return a reference to it:
```rust
div { "Value: {count()}" }
```

This will clone the `32` byte signal and then return a reference to the inner value:
```rust
div { "Value: {*(count.clone()).read()}" }
```

If your value is larger than 32 bytes, the second code snippet will save cloning that value. I also don't see how this fits into the managing props reference. If this were true, it would be better to document this in the state guide where signals are explained